src/app/router/Welcome/Welcome.tsx

```diff
@@ -53,6 +53,12 @@ function App() {
   const { t } = useTranslation(["welcome"]);
   const location = useLocation();
   const routesElement = useRoutes(routes);
+  //theme
+  useEffect(() => {
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
```

Comment:
why is this here but not on the other router/* ?

Reply:
This screen needed its own `useEffect` function because I could not use the one I wrote in `utils` for the other screens. The reason is that for the other screens I am checking the `Settings` for the theme selection. But on the Welcome screen the settings have not yet been initiated by the initial user on boarding. So in this case we only need to check what the users system theme settings are and not use the API call that doesn't exist yet. After the user sets up their first wallet they then have access to the Settings screen and can adjust the theme from there, which is why all of the screens after the initial on boarding check the Settings API.

Reply:
Does this make sense?

Reply:
hmm. yes makes sense. 
but does it hurt doing the settings call? When there is no special setting then it just uses the default system theme, doesn't it? 

having less code is always better - less things that break :)

Reply:
I think I tired using the same function for this screen but it wasn't working but I will t

> > I added this so that there was a smoother transition effect when the user switches between dark/light themes. It is not necessary though, I can remove it if you want?
> 
> I don't know. I just fear that this will be forgotten when developing new screens.

Yeah it's not a big deal, most people wouldn't notice it to be honest. I will remove.

Reply:
I think I tried this originally but it was not working - I will try again.

Reply:
So yes you are right about defaulting to system theme but for some reason when I first launch Alby this code does not work. There is an error in the devtools console, but when I refresh the page then it starts working. So I think in production it may also not work when you first launch Alby. Maybe it is because you cant use the API call right when Alby opens for the first time? Maybe @dylancom would know more but personally I think having one unique `useEffect` call here to get it working is not that much of an issue.

This is the error it gives:
`Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
    at wrappedSendMessageCallback (browser-polyfill.js:1187:1)`

Reply:
I think it means that `api.getSettings` does not exist yet when you first launch the extension.

Reply:
maybe we first need to debug and understand that error before drawing conclusions.

Reply:
Ok I did some more debugging and was able to add a new parameter to the function to check if it is on the initial Welcome page. If it is then it will run the code without the API call, all the other pages will use the API call like before.

Reply:
I really think we should first understand that error and then see how we can fix it instead of hacking around it. 
why is doing that call to load the settings a problem? I don't know the setup on this right now but I am wondering why the different pages/screens need to be treated differently. 

though it is great that it works, magic parameters make the code harder to read. people who don't have the background or in a few months don't know what the `true/false` means when reading `useThemeEffect(noDependencies, false);`

Reply:
The error was caused by onMessage listeners being registered after opening welcome.html. So I made sure that welcome is now opened after initialization has finished.